pkgs/compiler-test/tests/compiler/test/runtime.rkt

```diff
@@ -1,33 +1,53 @@
 #lang racket/base
 (require racket/system
+         racket/file
          compiler/find-exe)
 
 (define exe (find-exe))
 
-(define (try mode mod expect . extra-args)
+;; Return last line of the input, or #f if it's empty.
+(define (input-last-line input)
+  (for/fold ([prev #f]) ([s (in-lines input)])
+      (if (or (eof-object? s)
+              (equal? s "1 test passed"))
+          prev
+          s)))
+
+(define (try mode mod expect post-check . extra-args)
   (printf "trying ~s ~s\n" mod mode)
   (define s (open-output-bytes))
   (parameterize ([current-output-port s])
     (apply system* exe "-l-" "raco" "test"
            mode (append extra-args (list "-l" (string-append "tests/compiler/test/" mod)))))
-  (define last-line
-    (for/fold ([prev #f]) ([s (in-lines (open-input-bytes (get-output-bytes s)))])
-      (if (or (eof-object? s)
-              (equal? s "1 test passed"))
-          prev
-          s)))
-  (unless (equal? expect last-line)
-    (error 'runtime "test failed\n  module: ~s~a\n  expected: ~s\n  got: ~s"
+
+  (define last-line (input-last-line (open-input-bytes (get-output-bytes s))))
+  (define post-check-result (if (procedure? post-check) (post-check) #t))
+  (unless (and (equal? expect last-line)
+               post-check-result)
+    (error 'runtime "test failed\n  module: ~s~a\n  expected: ~s\n  got: ~s\n  post-check: ~s"
            mod
            (if (null? extra-args) "" (format "\n  extra args: ~s" extra-args))
            expect
-           last-line)))
+           last-line
+           post-check-result)))
+
+(define (output-file-check except file)
```

Comment:
Do you mean to name the variable `expect`?  

And again, it's probably better to close the port, or use `with-input-from-file` / `call-with-input-file*`.

Reply:
My bad :( Will be fixed in the new commit.